L15-PHP-Interceptors.md

```diff
@@ -0,0 +1,237 @@
+Title
+----
+* Author(s): ddyihai
+* Status: Draft
+* Implemented in: PHP
+
+## Abstract
+
+This proposal introduces the interceptor for gRPC PHP.
+__
+## Background
+
+The PHP language implementation for gRPC does not currently have an
+interceptor interface, whereas other languages like Go and Java do.
+Application concerns and extensibility are currently closed in the PHP
+implementation, making it difficult to extend the libraries for various
+application needs.
+
+
+### Related Proposals
+
+* [Python Client and Server Interceptors](https://github.com/grpc/proposal/pull/39)
+* [Ruby Client and Server Interceptors](https://github.com/grpc/proposal/pull/34)
+* [C# Client and Server Interceptors](https://github.com/grpc/proposal/pull/38)
+
+## Proposal
+
+Add client interceptors to the PHP gRPC implementation. Similar to the
+implementation in python, the channel is wrapper around an intercepted channel
+and the stub can use it the same way as the normal `Grpc\Channel`. RPCs dispatched 
+from such stubs then go through the interceptor chain for processing before they
+hit the underlying gRPC channel. 
+
+### Client Interceptors
+
+PHP provides a function `Grpc\Interceptor::intercept($channel, $interceptor)`
+to create a chain of interceptor, the API looks like below:
+ 
+```php
+Grpc\Interceptor::intercept(	// return intercepted channel
+	Channel				// underlying ‘Grpc\Channel’ or ‘Grpc\InterceptorChannel’
+	Grpc\Interceptor interceptor    // interceptors to add
+)
+```
+
+where `Grpc\InterceptorChannel` is the type returned by `Grpc\Interceptor::intercept`,
+who wrapper the `Grpc\Channel`. Since there is no private class in PHP, we can't prevent
+exposing it to the user.
+
+`Grpc\Interceptor` provides 4 methods for the user to override. In each method the user
+have access to the method, metadata and call options which are used to create the Call
```

Comment:
s/have/has.

The "method" on this line is different from "method" in the line above. Consider qualifying it as "RPC method" or something similar.